bluer-tools/src/blumon.rs

```diff
@@ -76,13 +160,85 @@ impl DeviceMonitor {
                         } else {
                             self.show_device(data).await;
                         }
+
+                        let device = self.adapter.device(data.address)?;
```

Comment:
Could you move the logging code into a separate method in AdvertisementLogger, which takes the device as argument?
Also note that the device could have disappeared after having been queried.